lib/cli.rb

```diff
@@ -83,6 +83,9 @@ def self.handle_error(e)
     Machinery::Ui.kill_pager
 
     case e
+    when OptionParser::AmbiguousOption
```

Comment:
Couldn't this be part of the exception handling in lines 92-97? That way we'd also show the hint about the `--help` option.